src/printer.cpp

```diff
@@ -171,6 +171,96 @@ void StrPrinter::bvisit(const Pow &x) {
     str_ = o.str();
 }
 
+//Polynomial printing, tests taken from SymPy and printing ensures that there is compatibility
+void StrPrinter::bvisit(const Polynomial &x) {
+    std::ostringstream s;
+    //bool variable needed to take care of cases like -5, -x, -3*x etc.
+    bool first = true;
+    //we iterate over the map in reverse order so that highest degree gets printed first
+    for (auto it = x.dict_.rbegin(); it != x.dict_.rend();) {
+        //given a term in polynomial, if coefficient is zero, print nothing
+        if (it->second == 0) {
+            //except when it is the only term, say "0"
+            if (it->first == 0) {
+                s << "0";
+            }
+            ++it;
+        }
+        //if the coefficient of a term is +1 or -1
+        else if (abs(it->second) == 1) {
+            //if exponent is 0, then print only coefficient
+            //in cases of -7, it is the only term, hence we print -7
+            //in cases of x - 7, the '-' is considered earlier, hence print only 7
+            if (it->first == 0) {
+                if (first) 
+                    s << it->second;
+                else 
+                    s << abs(it->second);
+            }
+            //if exponent is 1, print x instead of x**1 
+            else if (it->first == 1) {
+                //in cases of -x, print -x
+                //in cases of x**2 - x, print x, the '-' is considered earlier
+                if (first && it->second == -1) {
+                    s << "-" << x.var_->get_name();
+                }
+                else {
+                    s << x.var_->get_name();
+                }
+            }          
+            else {
+                if (first && it->second == -1) {
+                    s << "-" << x.var_->get_name() << "**"  << it->first;
+                }
+                else {
```

Comment:
Formatting `if` and `else` is documented here, https://github.com/sympy/symengine/blob/master/doc/style_guide.md
You have to fix it everywhere.